cmd/controller-manager/app/controller-manager.go

```diff
@@ -243,20 +243,30 @@ func getKubeFedConfig(opts *options.Options) *corev1b1.KubeFedConfig {
 	return fedConfig
 }
 
-func setDefaultKubeFedConfigScope(fedConfig *corev1b1.KubeFedConfig) {
-	if len(fedConfig.Spec.Scope) == 0 {
-		// TODO(sohankunkerkar) Remove when no longer necessary.
-		// This Environment variable is a temporary addition to support Red Hat's downstream testing efforts.
-		// Its continued existence should not be relied upon.
-		const defaultScopeEnv = "DEFAULT_KUBEFED_SCOPE"
-		defaultScope := os.Getenv(defaultScopeEnv)
-		if len(defaultScope) != 0 {
-			if defaultScope != string(apiextv1b1.ClusterScoped) && defaultScope != string(apiextv1b1.NamespaceScoped) {
-				klog.Fatalf("%s must be Cluster or Namespaced; got %q", defaultScopeEnv, defaultScope)
-			}
+func setDefaultKubeFedConfigScope(fedConfig *corev1b1.KubeFedConfig) bool {
+	// TODO(sohankunkerkar) Remove when no longer necessary.
+	// This Environment variable is a temporary addition to support Red Hat's downstream testing efforts.
+	// Its continued existence should not be relied upon.
+	const defaultScopeEnv = "DEFAULT_KUBEFED_SCOPE"
+	defaultScope := os.Getenv(defaultScopeEnv)
+	if len(defaultScope) != 0 {
+		if defaultScope != string(apiextv1b1.ClusterScoped) && defaultScope != string(apiextv1b1.NamespaceScoped) {
+			klog.Fatalf("%s must be one of %s or %s; got %q", defaultScopeEnv,
+				string(apiextv1b1.ClusterScoped), string(apiextv1b1.NamespaceScoped), defaultScope)
+			return false
+		}
+
+		if len(fedConfig.Spec.Scope) == 0 {
```

Comment:
This if/else is performing the same logic i.e. assigning `defaultScope` to `fedConfig.Spec.Scope` and returning `true`. We should either log when they don't match or remove the use of the if/else. I think logging may be preferred so there's some indication of the KubeFedConfig being overwritten with the value of `DEFAULT_KUBEFED_SCOPE`.

Reply:
@font, @xunpan made the changes as per the suggestion above.